pkg/ddevapp/apptypes.go

```diff
@@ -82,6 +82,8 @@ func init() {
 		AppTypeBackdrop: {
 			settingsCreator: createBackdropSettingsFile, uploadDir: getBackdropUploadDir, hookDefaultComments: getBackdropHooks, apptypeSettingsPaths: setBackdropSiteSettingsPaths, appTypeDetect: isBackdropApp, postImportDBAction: backdropPostImportDBAction, configOverrideAction: nil, postConfigAction: nil, postStartAction: backdropPostStartAction, importFilesAction: backdropImportFilesAction, defaultWorkingDirMap: docrootWorkingDir,
 		},
+		AppTypeMagento:  {},
```

Comment:
So no settings file management by ddev at this point, or anything like that? Might be OK. People like that stuff though.

Reply:
I have this install script that I would like to add as a custom command but I haven't yet figured out how I can do that for a specific app type (if it's currently possible at all):

```bash
#!/bin/bash

BOLD="\033[1m"
GRAY="\033[2;37m"
NC="\033[0m"

echo -en "${BOLD}Which package would you like to install?${NC} ${GRAY}(magento/project-community-edition)${NC} "

read package

package=package | tr -d "[:space:]"

if [ "" == "$package" ]
then
    package="magento/project-community-edition"
fi

composer create-project --repository=https://repo.magento.com/ $package ./install

echo "Copy files into ${WEBSERVER_DOCROOT}..."

shopt -s dotglob nullglob

mv ./install/* $WEBSERVER_DOCROOT/

$WEBSERVER_DOCROOT/bin/magento setup:install --base-url=https://$VIRTUAL_HOST --db-host=db --db-name=db --db-user=db --db-password=db --backend-frontname=admin --admin-firstname=Magento --admin-lastname=Admin --admin-email=admin@example.com --admin-user=admin --admin-password=password123 --language=en_US --currency=USD --timezone=America/Chicago --use-rewrites=1

```

Reply:
I think a postStartAction in apptypes.go (appTypeMatrix) could symlink an command from a .example and that might do what you want. Would be an interesting demonstration! 

Note that it's better to use `ddev composer create` instead of `composer create-project` probably because of the very unique problems Windows has with symlinks. 

Absolutely amazing pioneering on this. Thanks!